services/wireguard/service.go

```diff
@@ -27,35 +28,46 @@ import (
 	"github.com/mysteriumnetwork/node/identity"
 	"github.com/mysteriumnetwork/node/money"
 	dto_discovery "github.com/mysteriumnetwork/node/service_discovery/dto"
+	"github.com/mysteriumnetwork/node/services/wireguard/resources"
 	"github.com/mysteriumnetwork/node/session"
 )
 
 const logPrefix = "[service-wireguard] "
 
-// ErrAlreadyStarted is the error we return when the start is called multiple times
+// ErrAlreadyStarted is the error we return when the start is called multiple times.
 var ErrAlreadyStarted = errors.New("Service already started")
 
 // NewManager creates new instance of Wireguard service
 func NewManager(locationResolver location.Resolver, ipResolver ip.Resolver) *Manager {
 	return &Manager{
 		locationResolver: locationResolver,
 		ipResolver:       ipResolver,
+		resources:        resources.Handler{},
 	}
 }
 
-// Manager represents entrypoint for Wireguard service
+// Manager represents entrypoint for Wireguard service.
 type Manager struct {
-	process          sync.WaitGroup
-	locationResolver location.Resolver
-	ipResolver       ip.Resolver
-	isStarted        bool
+	isStarted          bool
+	process            sync.WaitGroup
+	locationResolver   location.Resolver
+	ipResolver         ip.Resolver
+	connectionEndpoint ConfigProvider
+	resources          resources.Handler
 }
 
-// Config represent a Wireguard service provider configuration that will be passed to the consumer for establishing a connection
+// Config represent a Wireguard service provider configuration that will be passed to the consumer for establishing a connection.
 type Config struct {
```

Comment:
I am still sticking to my previous suggestion -> ServiceConfiguration { 
    ProviderConfiguration,
    ConsumerConfiguration
}
if longer names better express itent and purpose - I go for them

Reply:
Changed it to serviceConfig. 

Looks like `Config` is more suitable for go then `Configuration`:
https://golang.org/search?q=Config
https://golang.org/search?q=Configuration